crates/bevy_scene/src/dynamic_scene_builder.rs

```diff
@@ -63,6 +66,18 @@ impl<'w> DynamicSceneBuilder<'w> {
         self.extract_entities(std::iter::once(entity))
     }
 
+    /// Despawns all enitities with no components.
+    ///
+    /// These were likely created because none of their components were present in the provided type registry upon extraction.
+    pub fn remove_empty_entities(&mut self) -> &mut Self {
+        self.extracted_scene = self
```

Comment:
Can this not use `HashMap::retain` instead of `collect`?

Reply:
It can!